frontend/src/metabase/querying/notebook/components/NotebookDataPicker/EmbeddingDataPicker/EmbeddingDataPicker.tsx

```diff
@@ -30,11 +37,19 @@ export function EmbeddingDataPicker({
   isDisabled,
   onChange,
 }: EmbeddingDataPickerProps) {
+  const userDefinedEntityTypes = useSelector(getEntityTypes);
+
   const { data: dataSourceCountData, isLoading: isDataSourceCountLoading } =
-    useSearchQuery({
-      models: ["dataset", "table"],
-      limit: 0,
-    });
+    useSearchQuery({ models: ["dataset", "table"], limit: 0 });
+
+  // We only count models to determine if we should hide tables.
+  // We don't need to do this if the user has defined their own visible entity types.
+  const { data: modelCountData, isLoading: isModelCountLoading } =
+    useSearchQuery(
+      userDefinedEntityTypes.length > 0
+        ? skipToken
+        : { models: ["dataset"], limit: 0 },
+    );
```

Comment:
- `limit: 0` means that we don't use the data, we only use the `total` count. same as the existing `useSearchQuery`.
- If the user provides their desired entity types, then we don't need to check the model count, so we can skip this request.